g2p/transducer/__init__.py

```diff
@@ -751,29 +756,37 @@ def apply_lexicon(self, to_convert: str):
             tg.output_string = ""
         else:
             tg.output_string = ""
-            edges: List[Tuple[Optional[int], Optional[int]]] = []
+            edges: List[Tuple[int, int]] = []
             in_pos = 0
             out_pos = 0
-            # Mappings are flat to save space
-            for n_inputs, outtxt in zip(alignment[::2], alignment[1::2]):
+            # Mappings are flattened to save space
+            for idx in range(0, len(alignment), 2):
+                (n_inputs, outtxt) = alignment[idx : idx + 2]
                 for i in range(n_inputs):
                     for j in range(len(outtxt)):
                         edges.append((in_pos + i, out_pos + j))
-                    if len(outtxt) == 0:  # Deletions
-                        edges.append((in_pos + i, None))
+                    if len(outtxt) == 0:
+                        # Attach deletions to the next input and the
+                        # previous output (fixed below if it does not
+                        # exist)
```

Comment:
This comment is confusing. By "next input", do you mean the current one being processed? Do you just mean "Attach deletions to the previous output"?

Reply:
Yes, the comment here is actually wrong.  Obviously the input for a deletion is just the deleted input.  But also it isn't actually attaching them to the previous output but rather the *next* one here.  The choice of previous or next is entirely arbitrary but probably it should use the previous one to be consistent with what rules do (even though in the current english lexicon this gives us considerably weirder alignments).